### PR TITLE
Migrate test for extension to functional tests

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

Migrate all the extensions tests that are reading files to the functional tests that permits to test more and are easier to understand.

Refactor prior to #5369

### CODE REVIEW

**Comment by DanielNoord** [comment]
Just FYI: This should also block #5349!

File: tests/functional/d/docstyle_extension.py [hunk 1]
```diff
@@ -1,35 +1,36 @@
...
         '''
 
-    def method2(self):
+    def method2(self):  # [bad-docstring-quotes]
         "bad docstring 1"
 
-    def method3(self):
-        'bad docstring 2'
+    def method3(self):  # [bad-docstring-quotes]
+        "bad docstring 2"
```
Comment (Pierre-Sassoulas) [inline tests/functional/d/docstyle_extension.py:28]:
```suggestion
        'bad docstring 2'
```

File: tests/functional/d/docstyle_extension.py [hunk 2]
```diff
@@ -1,35 +1,36 @@
...
 
-    def method4(self):
-        ' """bad docstring 3 '
+    def method4(self):  # [bad-docstring-quotes]
+        '"""bad docstring 3'
```
Comment (Pierre-Sassoulas) [inline tests/functional/d/docstyle_extension.py:31]:
```suggestion
        ' """bad docstring 3 '
```

**Review by cdce8p — COMMENTED** [review]
I'll leave that one up to you. It's definitely an improvement. (Just difficult to review the change.)

**Review by DanielNoord — COMMENTED** [review]
Yeah quite difficult to review this. I'm leaning to trusting you on this one, it's just tests anyway so no real danger of breaking things immediately.

**Comment by Pierre-Sassoulas** [comment]
Thank you for your trust 😄 But to explain better : As we did not change pylint's code the functional test are simply solidifying the current behavior of the code. The tests I removed were opening the file from ``tests/extension/data`` and asserting less than what the functional are asserting now. Two other files that appear as "removed" are actually one that I merged together (overlapping_exceptions). I checked that there was an actual output produced by the right expansion. If there is an issue in those it was preexisting. there are still some unittests for extansions in ``tests/extension``.

**Comment by cdce8p** [comment]
Just noticed something. We do have the `tests/functional/ext` folder. Maybe that would be a better place as it was meant for extensions.

**Comment by Pierre-Sassoulas** [comment]
> Maybe that would be a better place as it was meant for extensions.

Done, alongside some renaming, directory representing the name of the plugin creation and some name cleanup.

**Pull request merged**